src/api.ts

```diff
@@ -82,7 +82,9 @@ export interface SampleData {
 }
 
 export interface IndexedSampleData extends SampleData {
-  _index: number;
+  _flatIndex: number;
+  _templateIndex: number;
+  _template: TemplateName;
```

Comment:
After having read the rest of the diff, I would prefer the various `template` variables that contain template names, be named `templateName`; to distinguish them from the object stored in `TEMPLATES[templateName]`, which I assume is a "template".

Reply:
Yeah I like the idea. I started doing that and then realized that it would actually be _a lot_ of changes; enough that I'd be worried about breaking things. So I think I might save that as a follow-on task.

Reply:
https://github.com/microbiomedata/nmdc-field-notes/issues/209